docs/publish.md

```diff
@@ -50,11 +50,33 @@ The `main` repository is used by default.
 
 ## Authentication
 
-The first time you publish to a repository you need to authenticate using the `-u`/`--user` (environment variable `HATCH_INDEX_USER`) and `-a`/`--auth` (environment variable `HATCH_INDEX_AUTH`) options. You will be prompted if either option is not provided.
+As username the first source that defines one is used:
```

Comment:
What do you think about leaving this page mostly untouched and instead create a new `docs/how-to/publish/pypi.md`? Example: https://github.com/pypa/hatch/pull/1185

Context: https://github.com/pypa/hatch/issues/1123#issuecomment-1852328847

Reply:
my impression of this page as a whole is that it would be a good fit for the How-To pages. beside the small *Confirmation* section at the end, parts of the additions here are the only content that i would consider a reference rather than a guide.

my counter-proposal would be to move the verbosity of the *Artifact selection* and the *Repository* section in its entirety to a new How-To, move the configugration section where it belongs (or rather is missing, the config reference) and include pointers here in this introductory document?

Reply:
I like your proposal about emphasizing a config page though this page already represents the reference for the default publisher https://hatch.pypa.io/latest/plugins/publisher/package-index/ so I think we should add to that

Reply:
ah, thanks i missed that page.